openpype/pipeline/workfile/path_resolving.py

```diff
@@ -408,8 +408,12 @@ def get_custom_workfile_template(
     # add root dict
     anatomy_context_data["root"] = anatomy.roots
 
+    # extend anatomy context with os.environ
+    full_context_data = os.environ
+    full_context_data.update(anatomy_context_data)
```

Comment:
Ouch - this will actually update your local `os.environ` data. You'll need to work with a copy of the dict instead!

```suggestion
    full_context_data = os.environ.copy()
    full_context_data.update(anatomy_context_data)
```
OR:
```suggestion
full_context_data = dict(os.environ, **anatomy_context_data)
```

Reply:
good catch, i like the copy approach. merged in.